backend/compact-connect/lambdas/provider-data-v1/tests/function/test_data_model/test_client.py

```diff
@@ -138,9 +139,64 @@ def test_get_licenses_sorted_by_family_name_descending(self):
 
         # Verify sorting by family name (without getting into how duplicate family names are sorted)
         family_names = [item['familyName'] for item in resp['items']]
-        self.assertListEqual(sorted(family_names, reverse=True), family_names)
+        self.assertListEqual(sorted(family_names, key=quote, reverse=True), family_names)
 
-    def test_get_licenses_sorted_by_date_updated(self):
+    def test_get_providers_by_family_name(self):
+        from data_model.client import DataClient
+
+        # We'll provide names, so we know we'll have one record for our friends, Tess and Ted Testerly
+        self._generate_providers(
+            home='oh',
+            privilege='ne',
+            start_serial=9999,
+            names=(
+                ('Testerly', 'Tess'),
+                ('Testerly', 'Ted'),
+            )
+        )
+        client = DataClient(self.config)
+
+        resp = client.get_providers_sorted_by_family_name(  # pylint: disable=missing-kwoa
+            compact='aslp',
+            jurisdiction='oh',
+            provider_name=('Testerly', None),
+            scan_forward=False
+        )
+
+        self.assertEqual(2, len(resp['items']))
+        # Make sure both our providers have the expected familyName
+        for provider in resp['items']:
+            self.assertEqual('Testerly', provider['familyName'])
+
+    def test_get_providers_by_family_and_given_name(self):
+        from data_model.client import DataClient
+
+        # We'll provide names, so we know we'll have one record for our friends, Tess and Ted Testerly
+        self._generate_providers(
+            home='oh',
+            privilege='ne',
+            start_serial=9999,
+            names=(
+                ('Testerly', 'Tess'),
+                ('Testerly', 'Ted'),
+            )
+        )
+        client = DataClient(self.config)
+
+        resp = client.get_providers_sorted_by_family_name(  # pylint: disable=missing-kwoa
+            compact='aslp',
+            jurisdiction='oh',
+            # By providing given and family name, we can expect only one provider returned
+            provider_name=('Testerly', 'Tess'),
```

Comment:
Should/do we test anywhere that this throws an error if only first name is passed?

Reply:
Yeah, that's a good catch.